lib/money-rails/active_record/monetizable.rb

```diff
@@ -117,9 +117,19 @@ def monetize(*fields)
               memoized = instance_variable_get("@#{name}")
 
               # Dont create a new Money instance if the values haven't been changed.
-              if memoized && memoized.cents == amount &&
-                memoized.currency == attr_currency
-                result =  memoized
+              if memoized && memoized.cents == amount
+                if memoized.currency == attr_currency
+                  result =  memoized
+                else
+                  amount = memoized.amount
+                  unless amount.blank?
```

Comment:
You probably wanna extract this condition outside as well, because line 135 also uses it and it seems like no other action is happening if it doesn't pass